.github/workflows/e2e.yml

```diff
@@ -0,0 +1,80 @@
+name: Test e2e
+
+on:
```

Comment:
Following the existing pattern, can we have this workflow run via `on: workflow_call`?

The idea would be we add the workflow call to the PR and Publish workflows.
* https://github.com/voxel51/fiftyone/blob/develop/.github/workflows/pr.yml
* https://github.com/voxel51/fiftyone/blob/develop/.github/workflows/publish.yml

And add it to the `needs` lists so the workflows fail/reject when e2e tests fail
* https://github.com/voxel51/fiftyone/blob/327789be148f1f6c7fb0cb847db4c4f1b3f4447c/.github/workflows/pr.yml#L18
* https://github.com/voxel51/fiftyone/blob/327789be148f1f6c7fb0cb847db4c4f1b3f4447c/.github/workflows/publish.yml#L19
* https://github.com/voxel51/fiftyone/blob/327789be148f1f6c7fb0cb847db4c4f1b3f4447c/.github/workflows/publish.yml#L43